### PR TITLE
AO3-4750 Add link to most recent chapter in work blurbs

### PR DESCRIPTION
**Pull request by sarken**
## Issue

https://otwarchive.atlassian.net/browse/AO3-4750

## Purpose

Where it says "Chapters: [Current #]/[Expected #]" in the blurb stats line, the [Current #] should link to that chapter – so if it's "Chapters: 5/10", the 5 will take you directly to Chapter 5 of the work (http://test.ao3.org/works/###/chapters/###) 

## Testing

Refer to JIRA issue

## References

We're doing [AO3-5905](https://otwarchive.atlassian.net/browse/AO3-5905) instead of the below.

* ~Must be merged with pull request #2779 for issue AO3-4751~
* ~Must be deployed after the announcement post~

### CODE REVIEW

File: features/works/work_browse.feature [hunk 1]
```diff
@@ -8,3 +8,66 @@ Scenario: browsing works with incorrect page params in query string
...
+    And a fandom exists with name: "Canonical Fandom", canonical: true
+    And I post the 3 chapter work "WIP" with fandom "Canonical Fandom" with rating "Mature"
+  When I am logged out
+    And I browse the "Canonical Fandom" works
+    And I follow the recent chapter link for the work "WIP"
+  Then I should see "adult content"
+  When I follow "Proceed"
+  Then I should find myself on Chapter 3 of the work "WIP"
+  When "AO3-4751" is fixed
+    # And I should see "Hits: 1"
```
Comment (redsummernight) [inline features/works/work_browse.feature]: Noting that I'd rather merge this after #2779 so we don't need an extra pull to remove the TODO.

**Comment by redsummernight** [comment]
This has conflicts now.

File: features/step_definitions/work_steps.rb [hunk 2]
```diff
@@ -663,3 +680,9 @@
 Then /^the Remove Me As Chapter Co-Creator option should not be on the ([\d]+)(?:st|nd|rd|th) chapter$/ do |chapter_number|
   step %{I should not see "Remove Me As Chapter Co-Creator" within "ul#sortable_chapter_list > li:nth-of-type(#{chapter_number})"}
 end
+
+Then /^I should find myself on Chapter ([\d]+) of the work "([^\"]*)"$/ do |chapter_number, work_title|
+  step %{I should see "Chapter #{chapter_number}" within "div#chapters"}
```
Comment (redsummernight) [inline features/step_definitions/work_steps.rb]: Inside `div#chapters` there's `div#chapter-1` if you want to be even more specific.

Or we could add a path:

```rb
when /^chapter (\d+) of the work "(.*?)"/
  work = Work.find_by(title: $2)
  chapter = work.chapters_in_order(include_content: false)[$1.to_i - 1]
  work_chapter_path(work, chapter)
```

then use it like `Then I should be on chapter 2 of the work "WIP"`.

File: features/step_definitions/work_steps.rb [hunk 3]
```diff
@@ -663,3 +680,9 @@
 Then /^the Remove Me As Chapter Co-Creator option should not be on the ([\d]+)(?:st|nd|rd|th) chapter$/ do |chapter_number|
   step %{I should not see "Remove Me As Chapter Co-Creator" within "ul#sortable_chapter_list > li:nth-of-type(#{chapter_number})"}
 end
+
+Then /^I should find myself on Chapter ([\d]+) of the work "([^\"]*)"$/ do |chapter_number, work_title|
```
Comment (redsummernight) [inline features/step_definitions/work_steps.rb]: You can use `(\d+)` instead of `([\d]+)`, `(.*?)` instead of `([^\"]*)`.

Comment (sarken) [reply features/step_definitions/work_steps.rb]: I changed to `(.*?)` in #2779.

File: features/works/work_browse.feature (hunk 1 above)
Comment (redsummernight) [reply features/works/work_browse.feature]: Never mind, the other PR is based on this one. This one can go in first.

File: features/works/work_browse.feature [hunk 4]
```diff
@@ -38,3 +38,66 @@ Scenario: If works in a listing exceed the maximum search result count,
...
+    And an adult canonical rating exists with name: "Mature"
+    And a canonical fandom "Canonical Fandom"
+    And I post the 3 chapter work "WIP" with fandom "Canonical Fandom" with rating "Mature"
+  When I am logged out
+    And I browse the "Canonical Fandom" works
+    And I follow the recent chapter link for the work "WIP"
+  Then I should see "adult content"
+  When I follow "Proceed"
+  Then I should be on chapter 3 of the work "WIP"
+  When "AO3-4751" is fixed
```
Comment (redsummernight) [inline features/works/work_browse.feature:69]: This should be AO3-5905.

File: features/works/work_browse.feature [hunk 5]
```diff
@@ -38,3 +38,65 @@ Scenario: If works in a listing exceed the maximum search result count,
...
+  Given I am logged in as a random user
+    And an adult canonical rating exists with name: "Mature"
+    And a canonical fandom "Canonical Fandom"
+    And I post the 3 chapter work "WIP" with fandom "Canonical Fandom" with rating "Mature"
+  When I am logged out
+    And I browse the "Canonical Fandom" works
+    And I follow the recent chapter link for the work "WIP"
+  Then I should see "adult content"
+  When I follow "Proceed"
+    And the hit counts for all works are updated
```
Comment (redsummernight) [inline features/works/work_browse.feature:68]: If we're not testing hit counts, this step is unnecessary.

**Pull request merged**